src/components/Filter.js

```diff
@@ -1,59 +1,87 @@
-import React, { useState } from 'react'
+import React, { useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
 import styled from 'styled-components/macro'
 import { FilterList } from 'styled-icons/material/FilterList'
+import { CancelCircle } from 'styled-icons/icomoon/CancelCircle'
 
-export default function Filter ({activeTag, onTagClick, tags}) { 
-  const [isPopped, setIsPopped] = useState(false)
-  const [tagIsPopped, setTagIsPopped] = useState(false)
-  const [selectedFilter, setSelectedFilter] = useState('all')
-  
+Filter.propTypes = {
+  activeTag: PropTypes.string,
+  onTagClick: PropTypes.func,
+  tags: PropTypes.object
+}
+
+export default function Filter({ activeTag, onTagClick, tags }) {
+  const [isSelectorPopupOpen, setIsSelectorPopupOpen] = useState(false)
+  const [isFilterPopupOpen, setIsFilterPopupOpen] = useState(false)
+  const [selectedSelector, setSelectedSelector] = useState('all')
+
+  useEffect(() => {
+    isSelectorPopupOpen || setIsFilterPopupOpen(false)
+  }, [isSelectorPopupOpen])
 
-  function togglePopup(tag) {
-    if (selectedFilter === tag) {
-      setTagIsPopped(!tagIsPopped)
-      
-    }
-    else {
-      setSelectedFilter(tag)
-      setIsPopped(true)
-    }
+  function openFilterPopup(selector) {
+    setIsFilterPopupOpen(true)
+    setSelectedSelector(selector)
   }
-  function handleTagClick(event) {
-    event.stopPropagation()
-    setTagIsPopped(!tagIsPopped)
-    setIsPopped(!isPopped)
-    const clickedTag = event.currentTarget.textContent
-    onTagClick(selectedFilter, clickedTag )
+
+  function resetFilter() {
+    setSelectedSelector('all')
+    onTagClick('all')
+    setIsSelectorPopupOpen(false)
   }
 
-  function handleFilterClick() {
-    setIsPopped(!isPopped)
+  function handleTagClick(selectedSelector, tag) {
+    setIsSelectorPopupOpen(false)
+    setIsFilterPopupOpen(false)
```

Comment:
Das kannst du doch rausnehmen

Reply:
why? if I take it out the popUp doesn't close.

Reply:
If you take only 'setIsFilterPopupOpen(false)' out, then it should still work, because you have an effect on line 19